pysmt/fnode.py

```diff
@@ -121,13 +121,14 @@ def simplify(self):
         """Return a simplified version of the formula."""
         return _env().simplifier.simplify(self)
 
-    def substitute(self, subs):
+    def substitute(self, subs, interpretations=None):
         """Return a formula in which subformula have been substituted.
 
         subs is a dictionary mapping terms to be subtituted with their
         substitution.
```

Comment:
Can you extend the comment on interpretations? Something along the lines of:
```suggestion
        subs is a dictionary mapping terms to be subtituted with their
        substitution.
        interpretations is a dictionary mapping function symbols to an FunctionInterpretation objects describing the semantics of the function.
```